osbuild/util/runners.py

```diff
@@ -31,12 +33,38 @@ def sysusers():
         sys.exit(1)
 
 
+@contextmanager
+def create_machine_id_if_needed(tree=""):
+    """Create a machine-id with a fake machine id if it does not exist"""
```

Comment:
Would you mind adding a small additional line to the docstring like `If the machine-id did not exist before it will not be deleted but left empty` ? There is a subtle difference in the systemd world about missing entirely and empty so having a small comment/mention would be something I prefer :)

Reply:
Actually, I realized that we only need to keep the empty machine-id file if `tree != ""`, this is, when the function is called from the `org.osbuild.rpm` stage (to keep the semantics). If the function is called from the runner it is safe to remove it.

If it sounds reasonable to you I will update the script to keep the `machine-id` only when called from a stage and will update the docstring accordingly.

Reply:
Could we make the behaviour explicit with a `keep_empty=False` function argument?

Reply:
Yeah, that makes even more sense :D, will do that